pkg/build/multi.go

```diff
@@ -95,3 +96,40 @@ func (m *MultiArch) BuildLayers(ctx context.Context) (map[types.Architecture]v1.
 
 	return layers, errors.Join(errs...)
 }
+
+func (m *MultiArch) BuildPackageLists(ctx context.Context) (map[types.Architecture][]*apk.RepositoryPackage, error) {
+	var (
+		g  errgroup.Group
+		mu sync.Mutex
+	)
+	toInstalls := map[types.Architecture][]*apk.RepositoryPackage{}
+	errs := []error{}
+	for arch, bc := range m.Contexts {
+		arch, bc := arch, bc
+
+		g.Go(func() error {
+			toInstall, _, err := bc.apk.ResolveWorld(ctx)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("resolving apk packages for arch %q: %w", arch, err))
+			}
+
+			mu.Lock()
+			defer mu.Unlock()
+
+			if err != nil {
+				errs = append(errs, fmt.Errorf("for arch %q: %w", arch, err))
+				return nil
+			}
+
```

Comment:
This doesn't seem to do anything that the previous one didn't do?

Reply:
Main difference is happening in the constructor.